drgn/helpers/linux/mm.py

```diff
@@ -1435,3 +1437,123 @@ def in_direct_map(prog: Program, addr: IntegerLike) -> bool:
+    """
+    Return the task's resident set size (RSS) in pages
+
+    The task's RSS is the number of pages which are currently resident in
+    memory. The RSS values can be broken down into anonymous pages (not bound to
+    any file), file pages (those associated with memory mapped files), and
+    shared memory pages (those which aren't associated with on-disk files, but
+    belonging to shared memory mappings). This function returns a tuple
+    containing each category, but the common behavior is to use the "total"
+    value which sums them up.
+
+    :param task: `struct task_struct * for which to compute RSS
+    :returns: the file, anon, and shmem page values
+    """
+
+    # Kthreads have a NULL mm, simply skip them, returning 0.
+    if not task.mm:
+        return TaskRss(0, 0, 0, 0)
+
+    prog = task.prog_
+    rss_stat = task.mm.rss_stat
+
+    MM_FILEPAGES = prog.constant("MM_FILEPAGES").value_()
+    MM_ANONPAGES = prog.constant("MM_ANONPAGES").value_()
+    MM_SWAPENTS = prog.constant("MM_SWAPENTS").value_()
+
+    # This counter was added in Linux kernel commit eca56ff906bd ("mm, shmem:
+    # add internal shmem resident memory accounting") (in v4.5).
+    try:
+        MM_SHMEMPAGES = prog.constant("MM_SHMEMPAGES").value_()
+    except LookupError:
+        MM_SHMEMPAGES = -1
+
+    if rss_stat.type_.kind == TypeKind.ARRAY:
+        # Since v6.2, f1a7941243c10 ("mm: convert mm's rss stats into
+        # percpu_counter"), the "rss_stat" object is an array of percpu
+        # counters. Simply sum them up!
+        filerss = percpu_counter_sum(rss_stat[MM_FILEPAGES].address_of_())
+        anonrss = percpu_counter_sum(rss_stat[MM_ANONPAGES].address_of_())
+        swapents = percpu_counter_sum(rss_stat[MM_SWAPENTS].address_of_())
+        shmemrss = 0
+        if MM_SHMEMPAGES >= 0:
+            shmemrss = percpu_counter_sum(rss_stat[MM_SHMEMPAGES].address_of_())
+    else:
+        # Prior to this, the "rss_stat" was a structure containing counters that
+        # were cached on each task_struct and periodically updated into the
+        # mm_struct. We start with the counter values from the mm_struct and
+        # then sum up the cached copies from each thread.
+        filerss = rss_stat.count[MM_FILEPAGES].counter.value_()
+        anonrss = rss_stat.count[MM_ANONPAGES].counter.value_()
+        swapents = rss_stat.count[MM_SWAPENTS].counter.value_()
+        if MM_SHMEMPAGES >= 0:
+            shmemrss = rss_stat.count[MM_SHMEMPAGES].counter.value_()
+
+        for gtask in for_each_task_in_group(task, include_self=True):
+            filerss += gtask.rss_stat.count[MM_FILEPAGES].value_()
+            anonrss += gtask.rss_stat.count[MM_ANONPAGES].value_()
+            swapents += gtask.rss_stat.count[MM_SWAPENTS].value_()
+            if MM_SHMEMPAGES >= 0:
+                shmemrss += gtask.rss_stat.count[MM_SHMEMPAGES].value_()
+    rss = TaskRss(filerss, anonrss, shmemrss, swapents)
```

Comment:
There's no need for this variable assignment anymore, just return it directly.
```suggestion
    return TaskRss(filerss, anonrss, shmemrss, swapents)
```